### ISSUE TITLE
Ubuntu 18.04 node added to an existing Ubuntu 16.04 cluster causes network/stability issues

### ISSUE THREAD

**Issue by ufou** [kubernetes/kubernetes#94522]
**What happened**:
We have an existing bare-metal k8s cluster which (over the years) we've upgraded from v1.12 all the way to v1.18.5 using kubeadm. We have recently started experiencing performance issues which we believe might be fixed with more recent Linux kernels. So we drained & removed one of the existing cluster nodes and re-installed with Ubuntu 18.04 (fresh) after some basic configuration (same as it was before) of the node we then added it back to the cluster (same node name), after a few mins we started getting alerts via prometheus/alert manager from both Goldpinger and Mcrouter/memcached complaining that a node is down. At the same time if we run a ping command from the ubuntu 18.04 node to another node in the cluster we eventually see errors:
```
64 bytes from 10.0.1.1: icmp_seq=90 ttl=64 time=0.070 ms
64 bytes from 10.0.1.1: icmp_seq=91 ttl=64 time=0.088 ms
64 bytes from 10.0.1.1: icmp_seq=92 ttl=64 time=0.076 ms
ping: sendmsg: No buffer space available
ping: sendmsg: No buffer space available
ping: sendmsg: No buffer space available
```
If CTRL+C is issued on the above ping and the same ping command is run, the ping works again, but eventually goes back to the same state

If we run mtr (eg: `mtr --tcp -c 60 -4ra 10.0.1.1 10.0.1.5`) from other nodes in the cluster we see packet loss

The Ubuntu 18.04 server continues in this state until it's drained, removed from k8s and rebooted, at which point mtr and ping commands return back to normal...

If we rebuild the same server with Ubuntu 16.04, apply configuration and add it back to the cluster - everything is fine, no ping problems, mtr issues or noise from goldpinger/mcrouter alerts...

**What you expected to happen**:
I'd expect to be able to add an Ubuntu 18.04 node to our existing cluster without issue - what I'd like from anyone out there is some ideas on troubleshooting this, this problem has lingered with us - we first noticed this issue when we were running v1.16 k8s cluster and tried to upgrade a node to 18.04 and it's remained an issue through v1.17 and on to v1.18

**How to reproduce it (as minimally and precisely as possible)**:
I struggle to provide a minimal reproduction of this issue - I have tried with VMs (even mimicking our production private/public interfaces and configuration tweaks) - but I'm unable to trigger the same issue - I believe due to system load/amount of traffic flowing across the cluster...

**Anything else we need to know?**:
I've been debating where to report this - I thought about raising an issue on Ubuntu, SystemD or kube-router repos - but I figured the best starting place would be in this repo as it's only when the node is added to the k8s cluster that it starts to suffer problems. I actually tried running the Ubuntu 18.04 with every available Ubuntu LTS kernel but the problem remained, also at one point I thought I had narrowed this down to a SystemD bug after switching from systemd-networkd back to NetworkManager - but the problem still happened it was just one of the times when it took longer to show up.

Incidentally, we run Ubuntu 18.04 nodes for non-k8s workloads on the same network with similar configurations without these issues.

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.5", GitCommit:"e6503f8d8f769ace2f338794c914a96fc335df0f", GitTreeState:"clean", BuildDate:"2020-06-26T03:47:41Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.5", GitCommit:"e6503f8d8f769ace2f338794c914a96fc335df0f", GitTreeState:"clean", BuildDate:"2020-06-26T03:39:24Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

- Cloud provider or hardware configuration:
bare metal, perhaps most crucially we have dual NICs - one for private traffic and the other for public traffic (though this has not caused problems with a pure Ubuntu 16.04 cluster)

- OS (e.g: `cat /etc/os-release`):
All nodes in cluster:
```
NAME="Ubuntu"
VERSION="16.04.6 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.6 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```
Problem node:
```
NAME="Ubuntu"
VERSION="18.04.5 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.5 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```

- Kernel (e.g. `uname -a`):
all cluster nodes are:
```
Linux ubuntu1604worker 4.15.0-88-generic #88~16.04.1-Ubuntu SMP Wed Feb 12 04:19:15 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```
except for the single 18.04 node which seems to cause pain:
```
Linux ubuntu1804worker 5.4.0-45-generic #49~18.04.2-Ubuntu SMP Wed Aug 26 16:29:02 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

- Install tools: kubeadm
- Network plugin and version (if this is a network-related bug): kube-router
```
      - args:
        - --run-router=true
        - --run-firewall=true
        - --run-service-proxy=false
        - --metrics-port=8080
        - --metrics-path=/metrics
        - --bgp-graceful-restart=true
        env:
        - name: NODE_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: spec.nodeName
        - name: KUBE_ROUTER_CNI_CONF_FILE
          value: /etc/cni/net.d/10-kuberouter.conflist
        image: docker.io/cloudnativelabs/kube-router:v1.0.0
```

```
cat /etc/cni/net.d/10-kuberouter.conflist  | jq '.'
{
  "cniVersion": "0.3.0",
  "name": "mynet",
  "plugins": [
    {
      "bridge": "kube-bridge",
      "hairpinMode": true,
      "ipam": {
        "subnet": "10.244.0.0/24",
        "type": "host-local"
      },
      "isDefaultGateway": true,
      "name": "kubernetes",
      "type": "bridge"
    }
  ]
}
```
- Others:

**Comment by athenabot** [comment]
/triage unresolved

Comment `/remove-triage unresolved` when the issue is assessed and confirmed.

🤖 I am a bot run by vllry. 👩‍🔬

**Comment by jayunit100** [comment]
Intersting... 

- So, the *nodes* themselves, cannot ping each other.... 
- Are we then postulating here that *kubernetes* is causing network issues in your infrastructure ?
- What happens if you *uninstall* kubernetes from this cluster, does the network issue still persist? 

It seems like the errors your seeing are somehow related to ubuntu.  But your right it is interesting that you only see this when mixing 16 and 18.

- Do you think you can confirm for us wether or not removing k8s from this "cluster" is able to recover the ping functionality ?   
- Also, can you run the kubernetes e2e test suite and tell us which tests fail/pass on these mixed node clusters?

**Comment by ufou** [comment]
Thanks for your response!

> 
> * So, the _nodes_ themselves, cannot ping each other....
Not quite, high packet loss is introduced - TBH I've not actually run tests from 16.04 -> 16.04 k8s node while the 18.04 node is added to the cluster - I've been focused on 16.04 -> 18.04 and 18.04 -> 16.04 traffic flow - both TCP and ICMP appear affected

> * Are we then postulating here that _kubernetes_ is causing network issues in your infrastructure ?
It's where I'm at right now, purely because of testing various kernels and switching network daemons with the same results. I suspect it's some strange interaction with k8s, Ubuntu 18.04/systemd + our networking infrastructure...

> * What happens if you _uninstall_ kubernetes from this cluster, does the network issue still persist?
This is a production cluster so I cannot uninstall kubernetes - however we have 18.04 nodes on the same network that do not run kubernetes but do have heavy network IO work loads which run without issue. Also like I mentioned when the kubernetes 18.04 node is removed from the cluster and rebooted it does not have the same ping/mtr issues as it did when it was a member of the k8s cluster

> 
> It seems like the errors your seeing are somehow related to ubuntu. But your right it is interesting that you only see this when mixing 16 and 18.
> 
> * Do you think you can confirm for us wether or not removing k8s from this "cluster" is able to recover the ping functionality ?
Yes, removing the node from the cluster (and rebooting it) means ping / mtr fully recover...

> * Also, can you run the kubernetes e2e test suite and tell us which tests fail/pass on these mixed node clusters?
I will try this!

**Comment by ufou** [comment]
I couldn't get kubetest to work, also tried kubetest2 but had problems with that as well.

At the same time I was researching the ping error and noted that our net buffers were likely configured too low and were more appropriate for 1gb interfaces rather than our 10gb interfaces.

After applying the increased sysctl buffers to all nodes in the cluster (16.04 + the new 18.04 node) the problem appears to have been resolved. I cannot understand exactly why 16.04 seemed `OK` with this but 18.04 was not, one other option I noted was that we were setting `default_qdisc` to `fq` yet in 18.04 the default is `fq_codel` - we also reverted that...
